src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -170,23 +191,58 @@ int cb_add(LDAP *ld, Sockbuf *sb, LDAPURLDesc *srv, struct sockaddr *addr,
     int fd = -1;
     ldap_get_option(ld, LDAP_OPT_DESC, &fd);
     LOG(2) << "LDAP connect callback; file descriptor: " << fd;
-    static_cast<LDAPManagerImpl::ConnectionPoller*>(ctx->lc_arg)->start_poll(fd);
+    //static_cast<LDAPManagerImpl::ConnectionPoller*>(ctx->lc_arg)->start_poll(fd);
     return LDAP_SUCCESS;
 }
 
 void cb_del(LDAP *ld, Sockbuf *sb, struct ldap_conncb *ctx) {
     LOG(2) << "LDAP disconnect callback";
 }
 
+int rebindproc(LDAP* ld, const char* /* url */, ber_tag_t /* request */, ber_int_t /* msgid */, void* arg) {
+
+        const auto user = ldapGlobalParams.ldapQueryUser.get();
+        const auto password = ldapGlobalParams.ldapQueryPassword.get();
+
+        berval cred;
+        cred.bv_val = const_cast<char*>(password.c_str());
+        cred.bv_len = password.size();
+
+        return ldap_sasl_bind_s(ld, const_cast<char*>(user.c_str()), LDAP_SASL_SIMPLE, &cred,
+                               nullptr, nullptr, nullptr);
```

Comment:
We need to support "sasl" bind type too. See implementation in `LDAPBind`.

Reply:
Sasl code added.